### ISSUE TITLE
tab character support

### ISSUE THREAD

**Issue by FinzelStefan** [koalaman/shellcheck#1659]
As shellcheck can even be used for style checks I ask for an tabulator check in the verbose mode.

Sometimes tabulator characters especially in echo may lead to confusing output.

- a tabulator warning for strings in quotation marks would be fine.

- a general tabulator warning is not required, but would much more simple to implement. But this may lead to massive warnings as some use tabulator to indent source code.

**Comment by Gandalf-** [comment]
There doesn't seem to be a consensus on tab vs spaces use for shell scripts when used for indentation. I agree a general warning for tab usage doesn't seem like a good idea.
- https://unix.stackexchange.com/questions/39210/whats-the-standard-for-indentation-in-shell-scripts
- https://stackoverflow.com/questions/28832899/proper-indentation-of-bash-script

We could a general warning when a literal tab character is used in place of `\t` in quoted strings like you suggest. Making this verbose mode only behavior might be confusing.
```sh
#!/bin/bash

x=".<literal tab>a\t."
echo "$x"
```
```
$ bash example.sh
.       a       .
```

**Comment by Bearwolves** [comment]
It would be still nice to have the possibility to enable a check if tabs/enough spaces are used outside the echo or not i.e. in the "body" of our scripts. This check could be by default off.
This is especially useful in big teams to automate the check to avoid style conflict in software version control system :)

**Comment by mxcl** [comment]
to add my 2¢, I wanted a lint check for tab characters, not spaces since the heredoc syntax of `<<-HEREDOC` requires tabs and fails with spaces.
